cogs/committee_actions_tracking.py

```diff
@@ -0,0 +1,808 @@
+        required=True,
+        parameter_name="action_id",
+    )
+    async def delete_action(self, ctx: TeXBotApplicationContext, action_id: str) -> None:
+        """
+        Definition & callback response of the "delete" command.
+
+        Takes in an action as an argument and deletes it from the database.
+        This command should be used for administrative purposes only, in most circumstances
+        the update-status command should be used.
+        """
+        try:
+            action_id_int: int = int(action_id)
+        except ValueError:
+            await self.command_send_error(
+                ctx,
+                message="Action ID entered was not valid! Please use the autocomplete.",
+                logging_message=f"{ctx.user} entered action ID: {action_id} which was invalid",
+            )
+
+        try:
+            action: AssignedCommitteeAction = (
+                await AssignedCommitteeAction.objects.select_related().aget(id=action_id_int)
+            )
+        except (ObjectDoesNotExist, MultipleObjectsReturned):
+            await self.command_send_error(
+                ctx,
+                message="Action provided was either not unique or could not be found.",
+            )
+            return
+
+        await action.adelete()
+
+        await ctx.respond(content="Action successfully deleted.")
+
+
+class CommitteeActionsTrackingContextCommandsCog(CommitteeActionsTrackingBaseCog):
+    """Cog class to define the actions tracking message context commands."""
+
+    @discord.message_command(  # type: ignore[no-untyped-call, misc]
+        name="Action Message Author",
+        description="Creates a new action for the message author using the message content.",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def action_message_author(self, ctx: TeXBotApplicationContext, message: discord.Message) -> None:  # noqa: E501
+        """
+        Definition and callback response of the "action-message-author" message command.
+
+        Creates a new action assigned to the message author
+        using the message content as the description of the action.
+        """
+        if message.author.bot:
+            await ctx.respond(content="Actions cannot be assigned to bots you melon!")
+            logger.debug("User: %s, attempted to action a bot. silly billy.", ctx.user)
+            return
+
+        actioned_message_text: str = message.content
+        actioned_message_user: discord.Member | discord.User = message.author
+
+        if isinstance(actioned_message_user, discord.User):
+            await ctx.respond(content="Message author is not in the server!")
+            return
+
+        await self._create_action(
+            ctx,
+            actioned_message_user,
+            actioned_message_text,
+            silent=False,
+        )
+
+# NOTE: EVERYTHING BELOW THIS LINE MUST BE DELETED BEFORE MERGE
```

Comment:
I presume this was forgotten/a test for me

Reply:
yeah I just forgot :skull: